### ISSUE TITLE
Can the script change flow output,but do not change the original data？

### ISSUE THREAD

**Issue by apricotd** [mitmproxy/mitmproxy#4029]
#### Is your feature request related to a problem? Please describe.
I found it will change my original response data when i use a python script to resolve our compressed response data.The code below is my script code:

```import mitmproxy.http
from mitmproxy import ctx, http
import zlib


class DecompressData:

    def response(self, flow: mitmproxy.http.HTTPFlow):
        text = flow.response.get_content()

        text = zlib.decompress(text)


        text = str(text, encoding = "utf-8")

        flow.response.set_text(text)


addons = [
    DecompressData()
]
```

When the proxy return resolved data to my app,the decompress code will be excuted again in my app and decompress will failed.I do not want to change the code in my app.Is there exists some way that i can output the decompressd data and the proxy return the original(compressed) data?

#### Describe the solution you'd like
the flow can output some processed data but do not affect the proxy return package.

**Comment by mhils** [comment]
I'm not sure I understand what you want to achieve. Could you rephrase the problem you are trying to solve?
